hub/config/src/main/java/uk/gov/ida/hub/config/data/S3ConfigSource.java

```diff
@@ -5,55 +5,69 @@
 import com.amazonaws.services.s3.AmazonS3;
 import com.amazonaws.services.s3.AmazonS3ClientBuilder;
 import com.amazonaws.services.s3.model.GetObjectRequest;
-import com.amazonaws.services.s3.model.S3DataSource;
 import com.amazonaws.services.s3.model.S3Object;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import uk.gov.ida.hub.config.ConfigConfiguration;
 import uk.gov.ida.hub.config.domain.remoteconfig.RemoteConfigCollection;
 
-import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStream;
-import java.io.InputStreamReader;
 import java.text.DateFormat;
 import java.text.SimpleDateFormat;
 
-/**
- * Configure S3 connection here
- */
-public class S3ConfigSource{
-
-    private S3DataSource s3DataSource;
+public class S3ConfigSource {
+    private static final Logger LOG = LoggerFactory.getLogger(S3ConfigSource.class);
     private ConfigConfiguration configConfiguration;
     private AmazonS3 s3Client;
 
-    public S3ConfigSource(ConfigConfiguration configConfiguration){
+    public S3ConfigSource(ConfigConfiguration configConfiguration, AmazonS3 s3Client) {
         this.configConfiguration = configConfiguration;
-        BasicAWSCredentials awsCreds = new BasicAWSCredentials(configConfiguration.getSelfService().getS3AccessKeyId(),
-                configConfiguration.getSelfService().getS3SecretKeyId());
-        this.s3Client = AmazonS3ClientBuilder.standard().withRegion("eu-west-2")
-                .withCredentials(new AWSStaticCredentialsProvider(awsCreds))
-                .build();
+        this.s3Client = s3Client;
     }
 
-    public RemoteConfigCollection getRemoteConfig() throws IOException {
-        S3Object fullObject = s3Client.getObject(new GetObjectRequest(configConfiguration.getSelfService().getS3BucketName(),
-                configConfiguration.getSelfService().getS3ObjectKey()));
-        InputStream s3ObjectStream =  fullObject.getObjectContent();
-        BufferedReader reader = new BufferedReader(new InputStreamReader(s3ObjectStream));
-        StringBuilder configFile = new StringBuilder();
-        String line;
-        while ((line = reader.readLine()) != null) {
-            configFile.append(line);
+    /**
+     * TODO: Refactor this static method out with guice later
+     * @param configConfiguration
+     * @return
+     */
+    public static S3ConfigSource setupS3ConfigSource(ConfigConfiguration configConfiguration) {
```

Comment:
This isn't called from anywhere. As soon as we need to get one outside of a test we will wire it up in guice. I don't think we need this code.